### ISSUE TITLE
Can't add child entries to navigation

### ISSUE THREAD

**Issue by SteveEdson** [statamic/cms#4201]
## Bug Description
When trying to add a child navigation entry, I get the following error:

```
vue.common.prod.js:6 TypeError: Cannot read property 'linkExistingItem' of undefined
    at a.linkEntries (View.vue:274)
    at click (View.vue?dd60:9)
    at $e (vue.common.prod.js:6)
    at a.n (vue.common.prod.js:6)
    at $e (vue.common.prod.js:6)
    at a.e.$emit (vue.common.prod.js:6)
    at a.selectAndClose (DropdownItem.vue:33)
    at $e (vue.common.prod.js:6)
    at HTMLAnchorElement.n (vue.common.prod.js:6)
    at HTMLAnchorElement.Kr.o._wrapper (vue.common.prod.js:6)
```

## How to Reproduce
Add a navigation tree with the following config:

```yaml
title: 'Footer Menu'
```

There will only be an option to Add Nav Item, which will be a URL item, however, you also get the option to add a child entry. This will error. The following config, which allows selecting collections, will change the primary Add Nav Item button a dropdown, allowing adding URLs and Entries.

```yaml
title: 'Footer Menu'
collections:
  - pages
```

**In summary:** The main "add nav item" button has a check to see if entries can be added, but the check doesn't exist on child items, triggering the console error

## Extra Detail

<img width="350" alt="Screenshot 2021-08-31 at 21 20 43" src="https://user-images.githubusercontent.com/1201960/131570501-c6400211-bd0f-436c-b8ba-a393d3784f2a.png">
Add entry isn't available as an option on the button, but is an option on the entry.

## Environment

Statamic 3.2.1 Pro
Laravel 8.56.0
PHP 8.0.10

**Comment by jasonvarga** [comment]
Ah that `Add child link to entry` shouldn't be there because your nav doesn't have any collections configured (you can tell because the button at the very top says "Add Nav Item").

If you want to be able to add entries, go to the configuration for the nav and select the collection(s) you want to choose from.

If you don't want to be able to add entries, we'll need to fix it so that item doesn't show.

**Comment by SteveEdson** [comment]
Sorry yeah that's what I was getting at, and not explaining very well. Initially I thought the bug was that I couldn't add the entries, and got the console error, then I realised that I shouldn't be able to, because it's not been configured correctly.

Yes, in this instance, the "add child link to entry" in the context menu shouldn't be visible.

**Comment by jasonvarga** [comment]
Cool we're on the same page. 😄 
The bug is there's no check around the menu item.
